### ISSUE TITLE
Discussion: Opposed Rolls

### ISSUE THREAD

**Issue by curtisshipley** [Genii-Locorum/brp#4]
What were your thoughts about the UX for this? How were you thinking this might work in the UI? (I haven't check CoC or Runequest) to see how they do it.  The part I'm not sure about is how to correlate the two rolls together when they are potentially done by different players.

**Comment by Genii-Locorum** [comment]
I was thinking that on the first roll it checks to see if there is an open Opposed Roll - if not it creates it.  The roll results are stored in an array in the message data (currently results are stored but not in an array).  

On the second (or potentially subsequent rolls) the data is passed via socket to the GM (or go directly if its the GM directly) to the existing message which then gets updated and re-published.

The GM can close the opposed roll.

Does that help?

**Comment by curtisshipley** [comment]
Thanks. How are you thinking that a particular skill roll is marked as an opening opposed roll? Something like a checkbox on the message? 

I'm asking because some skills typically or commonly have opposed rolls, like sneak/stealth. Other skills may or may not have opposed rolls? For example: a character trying to pull a door open. In one case, the door may be old/stuck in the other case, there might be a goblin on the other side trying to hold it shut. The latter would require an opposed roll.

(You'll have to forgive if I'm missing something basic about BRP, I'm still coming up to speed.)

**Comment by Genii-Locorum** [comment]
When you roll a skill (unless you use the shift key) then you get a dialog box and I was going to put roll type as a drop down - with options for simple roll, opposed roll etc.  

If it's an opposed roll then I'll go through game.messages looking for an open opposed roll.  If there isn't one (or it's over a certain age) then it creates a new opposed message.  If there's an open opposed roll then the results get added to that.

For BRP as it's a framework that people then customise I'm not looking to code in sneak v stealth.  I'd leave it to the players/GM.  So in the case of sneak/stealth Player says I'm sneaking past the guard, GM make an oppossed stealth roll.  Player does that and GM does a Spot Hidden opposed roll for the guard - all results in the same message.  And I am not an expert on BRP - I took the Foundry implementation up as more of a challenge/puzzle :-)

**Comment by curtisshipley** [comment]
That makes sense. I'll take a look at that and see if I can get something working. We may want to revisit this because I can think of a case where someone rolling this skill may not (and should) not know that it is opposed. (Like someone trying to sneak and there happens to be a hidden guard.) That's kind of a corner case, so we can talk about that after I get the basic thing implemented.

**Comment by Genii-Locorum** [comment]
Heya - in the case you're talking about I as a GM would just ask a player for a roll and then I'd make a hidden roll and compare the results.  Now that then says there's another roll type/option of Hidden or GM Roll where dice aren't shown and the message is only visible to the GM.

And for clarity/transparency - I'm not a developer (I'd never touched Javascript before looking at Foundry) so apologies if any of the code is poorly written (and I keep forgetting the semi-colons).  There's probably lots that could be refactored - I've tended to reuse code I've got working across different systems and only later learn of a better way to do something.

And thank you for your help here.
